lib/usagereporter/usagereporter.go

```diff
@@ -243,6 +258,10 @@ func (r *UsageReporter[T]) Run(ctx context.Context) {
 		select {
 		case <-ctx.Done():
 			return
+		case <-r.gracefulStopScheduled:
+			r.enqueueBatch()
+			r.gracefulStopEnqueued <- true
```

Comment:
When using channels as a mechanism that something happened as opposed to exchange data, it's often better to close the channel than to write to it.

A write will only be observed by one reader, but a close will be observed by all readers.